packages/docusaurus-plugin-sitemap/src/__tests__/createSitemap.test.ts

```diff
@@ -39,6 +39,24 @@ describe('createSitemap', () => {
       'URL in docusaurus.config.js cannot be empty/undefined.',
     ));
 
+  it('invalid URL', () =>
+    expect(async () => {
+      // @ts-expect-error: test
+      await createSitemap(
+        {
+          url: 'example.com',
```

Comment:
The issue does not lie in the sitemap plugin. It's that this should not pass `validateConfig` at all. You need to edit `configValidation.ts` in core instead.

Reply:
In my latest commit, I updated my code to perform the error check in `configValidation.ts` and undo my previous modifications.